### PR TITLE
Reset schedule earlier to allow overlap with ggml graph computation on device

### PR DESCRIPTION
**Pull request by agray3**
Previously, significant CPU memset calls between each token generation were on critical path. 
This change performs these earlier, while the CPU is waiting for the previous token to be 
generated on the device. 

Refs #6763

### CODE REVIEW

**Comment by slaren** [comment]
The change looks good, I was aware that these `memset` are very expensive and I am working on other change to reduce the amount of memory that needs to be cleared. It may make more sense to make the reset call at the end of `llama_decode_internal` however.

**Comment by sorasoras** [comment]
It's this the reason the performance difference between a PCIE X2 P40 and PCIE X8 P40.

**Comment by agray3** [comment]
> The change looks good, I was aware that these `memset` are very expensive and I am working on other change to reduce the amount of memory that needs to be cleared. It may make more sense to make the reset call at the end of `llama_decode_internal` however.

Thanks, now moved the reset call as suggested.

**Comment by agray3** [comment]
> It's this the reason the performance difference between a PCIE X2 P40 and PCIE X8 P40.

This change doesn't affect any PCIe data transfers, only CPU activity. However if you have different CPUs or CPU memory configs in the two systems it could contribute to any difference.

File: ggml-backend.c [hunk 1]
```diff
@@ -1780,12 +1780,15 @@ void ggml_backend_sched_free(ggml_backend_sched_t sched) {
 
 void ggml_backend_sched_reset(ggml_backend_sched_t sched) {
     // reset state for the next run
-    size_t hash_size = sched->hash_set.size;
-    memset(sched->hash_set.keys,      0, sizeof(sched->hash_set.keys[0])     * hash_size); // NOLINT
-    memset(sched->tensor_backend_id, -1, sizeof(sched->tensor_backend_id[0]) * hash_size);
-    memset(sched->tensor_copies,      0, sizeof(sched->tensor_copies[0])     * hash_size);
+    if(!sched->is_reset)
+    {
```
Comment (slaren) [inline ggml-backend.c:1784]:
```suggestion
    if (!sched->is_reset) {
```

**Review by slaren — APPROVED** [review]

**Pull request merged**